### DISCUSSION TITLE
OverwriteModelError when using experimental-build-mode compile

### DISCUSSION THREAD

**Discussion by mattbolt** [payloadcms/payload#8639] [Q&A]
### Environment Info

Payload: 3.0.0-beta.110
Node: 18.20.4
Next.js: 15.0.0-canary.104

### Describe the Bug

When visiting the root url before visiting the /admin url and the database is empty, you get the following error:

``` 
⨯ OverwriteModelError: Cannot overwrite _pages_versions model once compiled.
    at Mongoose.model (/Users/***MASKED***/node_modules/mongoose/lib/index.js:587:13)
    at /Users/***MASKED***/.next/server/chunks/8914.js:90:18405
    at Array.forEach (<anonymous>)
    at Object.tK [as init] (/Users/***MASKED***/.next/server/chunks/8914.js:90:18042)
    at eD.init (/Users/***MASKED***/.next/server/chunks/8914.js:95:14493)
    at async eF (/Users/***MASKED***/.next/server/chunks/8914.js:95:16814) {
  digest: '2684875208'
}
```

### Reproduction Steps

Build the app using `output: "standalone"`, in `next.config.js` ... and compile using `next build --experimental-build-mode compile`.

If you subsequently visit the admin page after hitting the home page, you still get the error. However if you restart the server, the database gets initialised properly and you are asked to create the first user.

Also noting that if you call `next-build --experimental-build-mode generate` after the compile command and before visiting the root url, you do not get the above error.

### Adapters and Plugins

formBuilderPlugin, nestedDocsPlugin, redirectsPlugin, s3Storage, seoPlugin.

**Comment by jmikrut** [comment]
Hey @mattbolt I can't reproduce this. 

I just:

1. created a new blank project
2. ensured db was empty as you said
3. added a root page with a Payload local API call to get all users, so Payload would initialize at root url
4. built with output standalone, experimental mode compile
5. started server in prod
6. went to root
7. went to admin

Everything works well.

We will need a reproduction repo if this continues to happen for you but we're happy to get to the bottom of it if you can demonstrate that it is indeed an issue with Payload!

**Comment by andrei-ziminov** [comment]
Have the same issue. But I got it right after I redeployed on vercel.

```

GET

/admin

OverwriteModelError: Cannot overwrite `media` model once compiled.
    at Mongoose.model (/var/task/node_modules/mongoose/lib/index.js:587:13)
    at /var/task/.next/server/chunks/6315.js:156:17677
    at Array.forEach (<anonymous>)
    at Object.ew [as init] (/var/task/.next/server/chunks/6315.js:156:17234)
    at eQ.init (/var/task/.next/server/chunks/6315.js:156:36736) {
  digest: '3991473156'
}
```

**Comment by andrei-ziminov** [comment]
Hey @mattbolt,

I managed to fix the issue by upgrading Payload to `3.0.0-beta.115` along with all related packages. I also updated Next.js to `15.0.0-canary.196` and both React and React-DOM to `19.0.0-rc-3edc000d-20240926`.

Just a heads-up: if you go this route, you may need to make some adjustments in your Payload folder. For example, the importMap.js file was missing in my setup, so I had to add it manually. I followed the instructions in this repo for guidance: https://github.com/payloadcms/payload-3.0-demo.

**Comment by Stanisgrox** [comment]
Hello! I've came to the same issue with Payload v3.28.1
